File: src/Python/easy/array/problem485.py
from typing import List

class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        counter = 0
        max_consecutive = 0
        for i in range(len(nums)):
            counter = counter = counter + 1 if nums[i] == 1 else 0
            max_consecutive = max([counter, max_consecutive])
        return max_consecutive


# nums = [1,1,0,1,1,1]
nums = [1,0,1,1,0,1]

solution = Solution()
solution.findMaxConsecutiveOnes(nums)

File: src/JavaScript/easy/array/problem485.js
/**
 * @param {number[]} nums
 * @return {number}
 */

var findMaxConsecutiveOnes = function (nums) {
    let max = 0;
    let counter = 0;
    for (let i = 0; i < nums.length; i++) {
      if (nums[i] == 1) {
        counter++;
      } else {
        if (max < counter) {
          max = counter;
        }
        counter = 0;
      }
    }
    if (max < counter) {
      max = counter;
    }
    return max
  };
  
  let nums = [1, 1, 0, 1, 1, 1];
  // let nums = [1,0,1,1,0,1]
  
  findMaxConsecutiveOnes(nums);
  